### PR TITLE
Make make_cpio() take a list of files relative to the root directory

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
We operate on absolute paths all the time in kmod.py only for them to be made relative to the root directory just before they are passed to cpio. Let's save on the amount of allocations by always operating on paths relative to the root directory.

Because rglob() doesn't support returning paths relative to the given directory, we chdir() into the root directory before globbing instead.

### CODE REVIEW

File: mkosi/archive.py [hunk 1]
```diff
@@ -103,8 +103,8 @@ def make_cpio(
     sandbox: SandboxProtocol = nosandbox,
 ) -> None:
     if not files:
-        files = src.rglob("*")
-    files = sorted(files)
+        with chdir(src):
+            files = sorted(Path(".").rglob("*"))
```
Comment (behrmann) [inline mkosi/archive.py:107]: If files doesn't come in sorted already this the files in the cpio might not be sorted anymore and the whole thing less reproducible.

Comment (DaanDeMeyer) [reply mkosi/archive.py:107]: Yes but the only user of does does make sure the files are already sorted

Comment (behrmann) [reply mkosi/archive.py:107]: True, though it is something to keep in mind in addition. The overhead of sorting again should be negligible.

**Review by behrmann — APPROVED** [review]
Apart from the nit, LGTM.

**Pull request merged**